### PR TITLE
treewide: fix double quoted strings in one-liners

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Prior discussion: #108099
Authored by @kvtb

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
@SuperSandro2000 please run nixpkgs-review, bam has a hash change

File: pkgs/development/tools/build-managers/bam/default.nix [hunk 1]
```diff
@@ -13,9 +13,9 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ lua5_3 python ];
 
-  buildPhase = ''${stdenv.shell} make_unix.sh'';
+  buildPhase = "${stdenv.shell}make_unix.sh";
```
Comment (dotlambda) [inline pkgs/development/tools/build-managers/bam/default.nix:16]:
```suggestion
  buildPhase = "${stdenv.shell} make_unix.sh";
```

File: pkgs/development/tools/build-managers/bam/default.nix [hunk 2]
```diff
@@ -13,9 +13,9 @@ stdenv.mkDerivation rec {
...
 
-  checkPhase = ''${python.interpreter} scripts/test.py'';
+  checkPhase = "${python.interpreter}scripts/test.py";
```
Comment (dotlambda) [inline pkgs/development/tools/build-managers/bam/default.nix:18]:
```suggestion
  checkPhase = "${python.interpreter} scripts/test.py";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
This should only be merged if it doesn't cause any rebuilds.

File: pkgs/development/tools/build-managers/bam/default.nix (hunk 1 above)
Comment (kvtb) [reply pkgs/development/tools/build-managers/bam/default.nix:16]: oops.
stripping margin is not so easy

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110670` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>bam: log <a href="https://termbin.com/l40xj">https://termbin.com/l40xj</a></li>
  </ul>
</details>

**Comment by kvtb** [comment]
do not commit, there are more bugs.
I will rerun the script

**Comment by kvtb** [comment]
https://github.com/nixos/nixpkgs/commit/00ec1f7b62a052878904eedec097e389962c10eb.patch

**Comment by siraben** [comment]
By the way, how are you pushing these patches?

**Comment by kvtb** [comment]
> By the way, how are you pushing these patches?

There is @volth's Nix parser in Scala. Apparently, it has some bugs when processing double-quoted strings

**Comment by siraben** [comment]
> There is @volth's Nix parser in Scala. Apparently, it has some bugs when processing double-quoted strings 

Oh I mean pushing the patches under NixOS/nixpkgs directly, do you email or run a command etc.?

**Comment by kvtb** [comment]
> > There is @volth's Nix parser in Scala. Apparently, it has some bugs when processing double-quoted strings
> 
> Oh I mean pushing the patches under NixOS/nixpkgs directly, do you email or run a command etc.?

all patches are visible under NixOS/nixpkgs
yours too: https://github.com/nixos/nixpkgs/commit/b228182734ac783882249391d2b232b7ae9689cd.patch

**Pull request merged**